SendGrid/SendGrid/Resources/APIKeys.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Net.Http;
+
+namespace SendGrid.Resources
+{
+    public class APIKeysData
+    {
+        public string name { get; set; }
+    }
+
+    public class APIKeys
+    {
+        private string _endpoint;
+        private Client _client;
```

Comment:
I'm a little confused as to why compositionally, `APIKeys` holds a reference to the `Client`.